packages/functional-tests/pages/layout.ts

```diff
@@ -97,4 +97,58 @@ export abstract class BaseLayout {
       addEventListener('WebChannelMessageToChrome', listener);
     });
   }
+
+  /**
+   * Listens for a `WebChannelMessageToChrome` web channel event which
+   * occurs when we (web content) send a message to the browser.
+   *
+   * Responds with a `WebChannelMessageToContent` event containing event
+   * details passed in only when the given command matches the command from
+   * the listened-for event.
+   *
+   * @param webChannelMessage - Custom event details to send to the web content.
+   */
+  async respondToWebChannelMessage(webChannelMessage) {
+    const expectedCommand = webChannelMessage.message.command;
+    const response = webChannelMessage.message.data;
+
+    await this.page.evaluate(
+      ({ expectedCommand, response }) => {
+        function listener(e) {
+          const detail = JSON.parse((e as CustomEvent).detail);
+          const command = detail.message.command;
+          const messageId = detail.message.messageId;
+
+          if (command === expectedCommand) {
```

Comment:
@vbudhram I was able to remove the attached div stuff and it passes every time for me, it felt OK to remove. I can also remove some of this `if (command === expectedCommand) {` logic and it works, but, I'm not sure we should strip that out yet because I think we may need it in other tests if we need to wait for some event that doesn't fire immediately.